src/rkopenmdao/runge_kutta_scheme.py

```diff
@@ -179,3 +179,53 @@ def compute_step_transposed_jacvec(
     ) -> np.ndarray:
         """Currently not needed, keeping for later if necessary."""
         return new_state_perturbation + delta_t * stage_perturbation_field.sum(axis=0)
+
+
+class EmbeddedRungeKuttaScheme(RungeKuttaScheme):
+    """Implements functions used to apply an embedded Runge-Kutta method a function represented by
```

Comment:
I would merge that into the original `RungeKuttaScheme`. The new class should have a flag that determines if adaptive time stepping is requested and possible with the passed butcher tableau (i.e. if it is an embedded one). I would change the return value of the compute_step method from just the new state, to a tuple of (new_state, step_size, accepted) such that the new state, the new step size (if in an adapative scenario) and whether that step was accepted ( in non-adaptive case always true) can be returned.